### PR TITLE
Feature/broadcast notifications

### PR DESCRIPTION
**Pull request by lenchv**

### CODE REVIEW

File: resources/assets/js/app/services/BroadcastService.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Echo from 'laravel-echo';
+import Pusher from 'pusher-js';
+import {getAuthToken} from './AuthService';
+
+const PUSHER_KEY = '93ab4ad70ab16afc1f6d';
```
Comment (TatsuUkraine) [inline resources/assets/js/app/services/BroadcastService.js]: create config file where you will define such api keys

File: resources/assets/js/app/services/AuthService.js [hunk 2]
```diff
@@ -46,6 +46,14 @@ export const getAuthUser = (params) => {
     return data;
 };
 
+export const getUserId = () => {
+    if (isAuthorized()) {
+        return getAuthUser(['id']).user.id;
+    } else {
+        return -1;
```
Comment (TatsuUkraine) [inline resources/assets/js/app/services/AuthService.js]: return null

File: resources/assets/js/app/services/NotificationService.js [hunk 3]
```diff
@@ -13,3 +16,113 @@ export const markAsRead = (id) => {
 export const getCountUnread = () => {
     return securedRequest.get('/api/v1/notifications/unread/count');
 };
+
+export const getMessage = (notification) => {
+    const translate = LangService.translate;
+
+    switch (notification.type) {
+        case "booking_approved":
```
Comment (TatsuUkraine) [inline resources/assets/js/app/services/NotificationService.js]: define notification types as const

File: resources/assets/js/app/components/Notifications.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        Push.create(notification.title, {
+            body: notification.message,
+            timeout: notification.timeout,
+            onClick: function () {
+                notification.onClick();
+                window.focus();
+                this.close();
+            }
+        }).catch(() => {
+            this.setState({
```
Comment (TatsuUkraine) [inline resources/assets/js/app/components/Notifications.js]: move this state set to separate method

File: resources/assets/js/app/components/Notifications.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+    showNotification(notificationData) {
+        const messageData = getMessage(notificationData),
+            notification = {
+                key: notificationData.id,
+                message: messageData.title,
+                title: 'BeBeep',
+                timeout: 5000,
+                onClick: () => {
+                    browserHistory.push('/dashboard/notifications')
```
Comment (TatsuUkraine) [inline resources/assets/js/app/components/Notifications.js]: base on notification type navigate use to appropriate page like trip or booking page and mark this notification as read. For cases when page can't be opened - navigate to notification page

**Review by TatsuUkraine — CHANGES_REQUESTED** [review]

**Review by sergey-ben — DISMISSED** [review]

**Review by TatsuUkraine — APPROVED** [review]

**Pull request merged**